sdk/python/feast/feature_store.py

```diff
@@ -247,23 +247,28 @@ def list_feature_services(self) -> List[FeatureService]:
         """
         return self._registry.list_feature_services(self.project)
 
-    def list_feature_views(self, allow_cache: bool = False) -> List[FeatureView]:
+    def list_feature_views(
+        self, allow_cache: bool = False, tags: str = ""
+    ) -> List[FeatureView]:
         """
         Retrieves the list of feature views from the registry.
 
         Args:
+            tags: Tags to filter by.
             allow_cache: Whether to allow returning entities from a cached registry.
 
         Returns:
             A list of feature views.
         """
-        return self._list_feature_views(allow_cache)
+        return self._list_feature_views(allow_cache=allow_cache, tags=tags)
 
     def _list_feature_views(
         self,
         allow_cache: bool = False,
         hide_dummy_entity: bool = True,
+        tags: str = "",
```

Comment:
This comment is not specific to your change. WDYT of moving this method to BaseRegistry by adding some life cycle method to super class so that all the implementation of registries will invoke it. The goal is to keep feature store as simple delegator and keep it clean.

Reply:
i don't really have an opinion on it. if done though, i'd imagine it would have to be in a separate PR.

Reply:
I agree.

Reply:
That might be even more important when security model comes into picture (I don't think we have considered `list` permissions, though cc @dmartinol). Putting it in `BaseRegistry` means we can potentially have a way to allow giving permissions to `list` based on tags.

Reply:
+1

Reply:
+1 @tokoko we can discuss this in a different thread, but I assume that the LIST action is just equivalent to the READ one, applied to all the listed resources.
In short, if you can LIST objects [o1,...,oN], then you must also have permission to READ from any of them. If not, this means that the list API is returning resources that you're not authorized to READ.

Reply:
@dmartinol yeah, that was what I was getting at as well... Maybe if we extend registry API in the future to allow just listing the names of the objects w/o the objects themselves, then we might reconsider it, but right now that makes sense.

Reply:
@tokoko @lokeshrangineni the `tags` filtering logic has been moved to the `BaseRegistry` and module.